packages/vats/src/vat-transfer.js

```diff
@@ -0,0 +1,66 @@
+// @ts-check
+import { Far } from '@endo/far';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+
+import { provideLazy } from '@agoric/store';
+import { prepareVowTools } from '@agoric/vat-data/vow.js';
+import { prepareBridgeTargetModule } from './bridge-target.js';
+import { prepareTransferTools } from './transfer.js';
+
+export const buildRootObject = (_vatPowers, _args, baggage) => {
+  const zone = makeDurableZone(baggage);
+
+  const { makeBridgeTargetKit } = prepareBridgeTargetModule(
+    zone.subZone('bridge'),
+  );
+
+  const vowTools = prepareVowTools(zone.subZone('vow'));
+
+  const { makeTransferMiddleware } = prepareTransferTools(
+    zone.subZone('transfer'),
+    vowTools,
+  );
+
+  /**
+   * This 2-level structure is to avoid holding the bridge managers strongly, as
+   * well as accommodate the lack of complex keys.
+   *
+   * @type {WeakMapStore<
+   *   import('./types').ScopedBridgeManager,
+   *   MapStore<string, ReturnType<typeof makeBridgeTargetKit>>
+   * >}
+   */
+  const managerToKits = zone.weakMapStore('managerToHandler');
+  return Far('TransferVat', {
+    /**
+     * @param {import('./types').ScopedBridgeManager} manager
+     * @param {string} [inboundType]
+     */
+    provideBridgeTargetKit(manager, inboundType = 'IBC_EVENT') {
+      /** @type {MapStore<string, ReturnType<typeof makeBridgeTargetKit>>} */
+      const inboundTypeToKit = provideLazy(managerToKits, manager, () =>
+        zone.detached().mapStore('inboundTypeToKit'),
```

Comment:
Same question here, will `zone.detached()` cause upgrade problems? We should include a test of restarting this vat to document and validate expectations.

Reply:
No upgrade problems expected.  `provideLazy` checks `managerToKits` for a `manager` key, and if not existing, assigns a fresh mapStore to that key.  Because `managerToKits` is reachable via baggage, so is that mapStore.

If it weren't a `detached().mapStore`, then the `inboundTypeToKit` would be a singleton for the `zone`.